content/blog/zardshard/2023-08-18_[gsoc_2023]_improving_icon-o-matic_final_report.md

```diff
@@ -0,0 +1,92 @@
+
+**Commits:**
+* [Icon-O-Matic: Add reference images](https://git.haiku-os.org/haiku/commit/?id=098eaec6305ae804d3eb6c8e6c6aad790fb4cfb1)
+* [Icon-O-Matic: Save alpha of reference images](https://git.haiku-os.org/haiku/commit/?id=4ed150bdc36d063144a3552305c5d77ecac88427) (bugfix)
+
+### Added perspective transformations
+
+![Screenshot of a grid being put into perspective](/files/blog/zardshard/perspective_transformer.png)
+
+I believe this feature will be a lot less useful than reference images. Ironically, it also took quite a bit more time to implement than reference images. Implementing this feature also took the most math. I used quite a bit of linear algebra to figure out how to determine whether a perspective transformation was valid or not!
+
+**Commits:**
+* [agg: Pull in updated perspective transformation](https://review.haiku-os.org/c/haiku/+/6808) (unmerged as of writing)
+* [Icon-O-Matic: Add perspective transformations](https://review.haiku-os.org/c/haiku/+/6801) (unmerged as of writing)
+
+### Reduced repeated code
+
+This shaved off 1,300 lines of code from Icon-O-Matic. This one is the only change that caused a regression, at least, as far as I'm aware ;)
+
+**Commits:**
+
+* [Icon-O-Matic: Generalize some classes](https://git.haiku-os.org/haiku/commit/?id=6427935280aaac0a1a4557bae55184708819560d)
+* [libicon: Notify Shape on transformer addition/removal](https://git.haiku-os.org/haiku/commit/?id=c5abd6a796982af7cc456f832086da2e4a13ad83) (regression fix)
+
+### Miscellaneous improvements
+
+There are various miscellaneous improvements that I made to Icon-O-Matic along the way. Some are bugfixes. Others introduce some simple features. None of these took very much work.
+
+**Commits:**
+
+* [Icon-O-Matic: Remove unused function parameter](https://git.haiku-os.org/haiku/commit/?id=8ffe04780def2570c3806e6c18fa14a79fd105eb)
+* [Icon-O-Matic: Remove unused dependency](https://git.haiku-os.org/haiku/commit/?id=16edf24a3c77400a81110adb8933f640bf210089)
+* [Icon-O-Matic: Remove dead homebrew translation system](https://git.haiku-os.org/haiku/commit/?id=a75a222b35d17cd83bc75253f3cd8e24a6a911f4)
+* [Icon-O-Matic: Make left sidebar lists resizable](https://git.haiku-os.org/haiku/commit/?id=c70aca1135bb1e1999db080457a40e9d36989582)
+* [Icon-O-Matic: Fix window resizing bug](https://git.haiku-os.org/haiku/commit/?id=6da29a769f1e2afb5d98f0470b35497d7f3ef3ea)
+* [Icon-O-Matic: Improve saving-related error messages](https://git.haiku-os.org/haiku/commit/?id=405c9dc3f27cf660facae5253303ea24aaaf382e)
+* and various others
+
```

Comment:
the work on upstreaming our agg changes (and researching the different forks) deserves a mention here, even if the result is not a commit in Haiku repository.

In fact, I think that is also part of the GSoC experience: you get to be a contributo to Haiku, but also in the broader open source world and its many projects.